js/modules/loki_app_dot_net_api.js

```diff
@@ -291,6 +294,86 @@ class LokiAppDotNetServerAPI {
     }
   }
 
+  static async _sendToProxy(
+    fetchOptions,
+    endpoint,
+    method,
+    { ephemeralKey, symKey, iv }
+  ) {
+    const randSnode = await lokiSnodeAPI.getRandomSnodeAddress();
+    const url = `https://${randSnode.ip}:${randSnode.port}/file_proxy`;
+
+    const payloadObj = {
+      // I think this is a stream, we may need to collect it all?
+      body: fetchOptions.body, // might need to b64 if binary...
+      endpoint,
+      method,
+      headers: fetchOptions.headers,
+    };
+
+    // from https://github.com/sindresorhus/is-stream/blob/master/index.js
+    if (
+      payloadObj.body &&
+      typeof payloadObj.body === 'object' &&
+      typeof payloadObj.body.pipe === 'function'
+    ) {
+      log.info('detected body is a stream');
+      const fData = payloadObj.body.getBuffer();
+      const fHeaders = payloadObj.body.getHeaders();
+      // update headers for boundary
+      payloadObj.headers = { ...payloadObj.headers, ...fHeaders };
+      // update body with base64 chunk
+      payloadObj.body = {
+        fileUpload: fData.toString('base64'),
+      };
+    }
+
+    // convert our payload to binary buffer
+    const payloadData = Buffer.from(
+      dcodeIO.ByteBuffer.wrap(JSON.stringify(payloadObj)).toArrayBuffer()
+    );
+    payloadObj.body = false; // free memory
+
+    // encrypt payloadData with symmetric Key using iv
+    const cipherBody = await libsignal.crypto.encrypt(symKey, payloadData, iv);
```

Comment:
`libloki.crypto.DHEncrypt` will do all this

Reply:
It doesn't return the IV, I'd have to parse it out. Or I'll need to refactor it to take an IV passed in.

Reply:
Where do you need the IV after you get `ivAndCiphertext`?

Reply:
the decoding the response since I reuse the IV